src/anchored-position.ts

```diff
@@ -440,3 +476,23 @@ function shouldRecalculatePosition(
     )
   }
 }
+
+/**
+ * Determines if there is an overflow
+ * @param align,
+ * @param currentPos
+ * @param containerDimensions
+ * @param elementDimensions
+ */
+function shouldRecalculateAlignment(
+  align: AnchorAlignment,
+  currentPos: Position,
+  containerDimensions: BoxPosition,
+  elementDimensions: Size
+) {
+  if (align === 'end') {
+    return currentPos.left < containerDimensions.left
+  } else if (align === 'start' || align === 'center') {
+    return currentPos.left + elementDimensions.width > containerDimensions.left + containerDimensions.width
+  }
```

Comment:
oooh that's a great point! I understand what you mean, will play around with it a bit more

Reply:
I tried to force the above example, does this feel like the correct behavior?

It stays `center` as long as it can, then switches to `end` if there isn't enough space.


https://user-images.githubusercontent.com/1863771/162779652-e6a8df26-7b77-4b4a-a576-03a2ebf3cb88.mov

Reply:
> Notably, this logic for determining overflow may need to consider whether anchorSide is top or bottom, rather than left or right.

Not sure I understand when this breaks. If the anchorSide is `right` and there isn't enough space, it switches to `left` before we re-evaluate alignment. 🤔  What scenario am I missing?

Reply:
I agree with you. I don't think `shouldRecalculateAlignment` would ever be called for the `anchorSide` left or right scenario 👍 If I'm understanding correctly, this method is only relevant for `anchorSide` top or bottom.

Reply:
Yep! It's only relevant if `anchorSide` is top or bottom 👍